Add Contextual VideoCard and VideoThumbnail components (#2725)
Shopify/polaris-react: src/components/VideoCard/README.md

---
name: Video card
category: Structure
keywords:
  - VideoCard
  - actionable
  - updates
  - new features
  - video card
  - feature card
  - card with thumbnail
  - thumbnail card
---

# Video card

Video cards provide a consistent layout for contextual learning system content. Use to wrap thumbnails of educational videos about Shopify admin features in context.

---

## Best practices

Video cards should:

- Clearly articulate the benefit of the feature and what it does in the context of where it is managed
- Provide merchants with a clear call to action
- Show targeted content toward specific user types where possible to maximize relevance and impact

> - Use a video player with a thumbnail that helps to communicate the subject of the video or merchant benefit

Reviewer: ```suggestion
- Wrap a [video thumbnail](https://polaris.shopify.com/components/images-and-icons/video-thumbnail) that communicates the subject of the video or merchant benefit
```

- Be positioned at the bottom of the page to avoid getting in the way of a merchant task, unless used within an empty state
- Be dismissable

---

## Content guidelines

> - Do not use video cards as advertisements for your feature. Instead they should educate the merchant about how to accomplish tasks related to the section they’re in.

Reviewer: ```suggestion
- Video cards should not serve as advertisements for your feature. Use video cards to educate merchants about how to accomplish tasks related to the section they’re in.
```

Add a menu item

<!-- end -->

### Thumbnail

- Relevant: should describe the video through the use of images or be an image of a section in the video
- Follow the 16:9 ratio, otherwise the image may appear cropped
- If the thumbnail shows a person, avoid cropping the person’s head off

---

## Examples

### Basic video card

Use to surface educational information about a feature or opportunity in the context of where it is managed.

```jsx
<VideoCard
  title="Getting Started"
  primaryAction={[
    {
      content: 'Learn about getting started',
      onAction: () => {},
    },
  ]}
  description="Discover how Shopify can power up your entrepreneurial journey."
  popoverActions={[{content: 'Dismiss', onAction: () => {}}]}
  portrait
>
  <VideoThumbnail
    videoLength={80}
    thumbnailUrl="https://burst.shopifycdn.com/photos/smiling-businesswoman-in-office.jpg?width=1850"
  />
</VideoCard>
```

### Video card with secondary action

Use when there are two distinct actions merchants can take on the information in the video.

```jsx
<VideoCard
  title="Getting Started"
  primaryAction={{
    content: 'Learn about getting started',
    onAction: () => {},
  }}
  secondaryAction={{

>     content: 'Click here',

Reviewer: @krismausser @raydg-shopify is there a real example in the works or in production of a video card with a secondary action? If not, it's okay for there not to be an example for this prop, its shape is documented in the props table of the doc.

```suggestion
    content: 'Some real action here',
```
